File: code/933.number-of-recent-calls.php
<?php
/*
 * @lc app=leetcode id=933 lang=php
 *
 * [933] Number of Recent Calls
 */

// @lc code=start
class RecentCounter {
    private array $queues = [];
    /**
     */
    function __construct() {
    }

    /**
     * @param Integer $t
     * @return Integer
     */
    function ping($t) {
        $this->queues[] = $t;
        foreach ($this->queues as $key => $time) {
            if ($time >= $t - 3000) {
                break;
            }
            unset($this->queues[$key]);
        }
        return count($this->queues);
    }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * $obj = RecentCounter();
 * $ret_1 = $obj->ping($t);
 */
// @lc code=end

File: code/933.number-of-recent-calls.go
package code

/*
 * @lc app=leetcode id=933 lang=golang
 *
 * [933] Number of Recent Calls
 */

// @lc code=start
type RecentCounter struct {
	queues []int
}

func Constructor() RecentCounter {
	return RecentCounter{}
}

func (this *RecentCounter) Ping(t int) int {
	this.queues = append(this.queues, t)

	for this.queues[0] < t-3000 {
		this.queues = this.queues[1:]
	}

	return len(this.queues)
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Ping(t);
 */
// @lc code=end
